accord-core/src/test/java/accord/impl/basic/DelayedExecutorService.java

```diff
@@ -0,0 +1,107 @@
+package accord.impl.basic;
+
+import accord.burn.random.Decision;
+import accord.burn.random.IntRange;
+import accord.burn.random.RandomInt;
+import accord.burn.random.SegmentedIntRange;
+import org.apache.cassandra.concurrent.FutureTask;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.AbstractExecutorService;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+public class DelayedExecutorService extends AbstractExecutorService
```

Comment:
Perhaps make clearer this isn't a real executor service? Perhaps `SimulatedExecutorService` or some other better name.

Reply:
used `SimulatedDelayedExecutorService` to denote both semantics; simulated + delayed